oauthenticator/globus.py

```diff
@@ -166,7 +166,10 @@ def authenticate(self, handler, data=None):
         # Doing the code for token for id_token exchange
         tokens = client.oauth2_exchange_code_for_tokens(code)
         id_token = tokens.decode_id_token(client)
-        username, domain = id_token.get('preferred_username').split('@')
```

Comment:
We can use `username, domain = id_token.split('@', 1)` to split only on the first occurrence of `@`.

Reply:
I like that much better, thanks!